tools/clang/unittests/HLSL/CompilerTest.cpp

```diff
@@ -2589,6 +2602,79 @@ TEST_F(CompilerTest, CompileWhenVdThenProducesDxilContainer) {
   VERIFY_IS_TRUE(hlsl::IsValidDxilContainer(reinterpret_cast<hlsl::DxilContainerHeader *>(pResultBlob->GetBufferPointer()), pResultBlob->GetBufferSize()));
 }
 
+void CompilerTest::TestEncodingImpl(const void *sourceData, size_t sourceSize, UINT32 codePage, const void *includedData, size_t includedSize,
+                                    const WCHAR *encoding) {
+  CComPtr<IDxcCompiler> pCompiler;
+  CComPtr<IDxcOperationResult> pResult;
+  CComPtr<IDxcBlobEncoding> pSource;
+  CComPtr<TestIncludeHandler> pInclude;
+  VERIFY_SUCCEEDED(CreateCompiler(&pCompiler));
+  CreateBlobPinned((const char *)sourceData, sourceSize, codePage,
+                   &pSource);
+  pInclude = new TestIncludeHandler(m_dllSupport);
+  pInclude->CallResults.emplace_back(includedData, includedSize, 0);
```

Comment:
I think I'd like to understand if this include data is deliberately being left without explicit encoding to ensure that the flag forces it to be interpreted accordingly.

Reply:
Ok that makes sense to switch to CP_ACP for clarity. I just did some additional testing and the parameter value doesn't make a difference (whether it is 0, CP_ACP, or completely omitted).

Reply:
Yes, it's using 0/CP_ACP because it's explicitly testing the default encoding option for when we just have bytes from a file and don't know the encoding.